crates/fuel-core/src/graphql_api/indexation/balances.rs

```diff
@@ -0,0 +1,598 @@
+use fuel_core_storage::StorageAsMut;
+use fuel_core_types::{
+    entities::{
+        coins::coin::Coin,
+        Message,
+    },
+    services::executor::Event,
+};
+
+use crate::graphql_api::{
+    ports::worker::OffChainDatabaseTransaction,
+    storage::balances::{
+        CoinBalances,
+        CoinBalancesKey,
+        MessageBalance,
+        MessageBalances,
+    },
+};
+
+use super::error::IndexationError;
+
+fn increase_message_balance<T>(
+    block_st_transaction: &mut T,
+    message: &Message,
+) -> Result<(), IndexationError>
+where
+    T: OffChainDatabaseTransaction,
+{
+    let key = message.recipient();
+    let storage = block_st_transaction.storage::<MessageBalances>();
+    let current_balance = storage.get(key)?.unwrap_or_default().into_owned();
+    let MessageBalance {
+        mut retryable,
+        mut non_retryable,
+    } = current_balance;
+    if message.is_retryable_message() {
+        retryable = retryable.saturating_add(message.amount() as u128);
+    } else {
+        non_retryable = non_retryable.saturating_add(message.amount() as u128);
+    }
+    let new_balance = MessageBalance {
+        retryable,
+        non_retryable,
+    };
+
+    block_st_transaction
+        .storage::<MessageBalances>()
+        .insert(key, &new_balance)
+        .map_err(Into::into)
+}
+
+fn decrease_message_balance<T>(
+    block_st_transaction: &mut T,
+    message: &Message,
+) -> Result<(), IndexationError>
+where
+    T: OffChainDatabaseTransaction,
+{
+    let key = message.recipient();
+    let storage = block_st_transaction.storage::<MessageBalances>();
+    let MessageBalance {
+        retryable,
+        non_retryable,
+    } = storage.get(key)?.unwrap_or_default().into_owned();
+    let current_balance = if message.is_retryable_message() {
+        retryable
+    } else {
+        non_retryable
+    };
+
+    let new_amount = current_balance
+        .checked_sub(message.amount() as u128)
```

Comment:
Same here
```suggestion
        .checked_sub(u128::from(message.amount()))
```

Reply:
Updated in 5eb9bb06309ef064a281043fc32a8aaffc916989